routers/web/repo/pull.go

```diff
@@ -575,7 +577,9 @@ func PrepareViewPullInfo(ctx *context.Context, issue *issues_model.Issue) *git.C
 	if pb != nil && pb.EnableStatusCheck {
 		ctx.Data["is_context_required"] = func(context string) bool {
 			for _, c := range pb.StatusCheckContexts {
-				if c == context {
+				if gp, err := glob.Compile(c); err != nil {
+					log.Error("glob.Compile %s failed. Error: %v", c, err)
```

Comment:
It would cause a lot of noises for site admins if users make a lot of incorrect glob expressions.

Reply:
> It would cause a lot of noises for site admins if users make a lot of incorrect glob expressions.

They have been checked on frontend.

Reply:
```suggestion
```

Yup, so I think these errors should be ignored.

Reply:
Since they have been checked, this will be come a rare event. So maybe recording them is better?

Reply:
Why server log should show user's error?

Reply:
We reuse `status_check_contexts` here. We can check the patterns that users newly entered but we can't confirm whether the user's previous input is a valid glob. So I think keeping this log can help us troubleshoot such problems.

Reply:
I believe every line of code should be useful and helpful.

Think about two user stories:

1. Story 1
    * If a user see their commit status doesn't work as expected, they login to Gitea server and check error logs.
    * If a site admin have thousands of repos, they would see a lot of "glob.Compile %s failed. Error: %v"
2. Story 2
    * If a user see their commit status doesn't work as expected, they go to repo setting page and check whether the rules are correct.


If you do `log.Error` here, then that's Story 1, I do not think it is reasonable.

If story 1 is not reasonable, then this `log.Error` is not useful.

Reply:
That's right. This error log seems not useful and was removed in 8e190b0